src/landscape/landscapeService.test.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2023 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+import * as terrasoApiToMock from 'terrasoApi/terrasoBackend/api';
+
+import * as landscapeService from 'landscape/landscapeService';
+
+jest.mock('terrasoApi/terrasoBackend/api');
+const terrasoApi = jest.mocked(terrasoApiToMock);
+
+test('LandscapeService: Fetch landscape with missing fields', async () => {
+  terrasoApi.requestGraphQL.mockReturnValue(
+    Promise.resolve({
+      landscapes: {
+        edges: [
+          {
+            node: {
+              name: 'Landscape name',
+              description: 'Landscape description',
+              website: 'https://www.landscape.org',
+            },
+          },
+        ],
+      },
+    })
+  );
+  const landscape = await landscapeService.fetchLandscapeToView('');
+  expect(landscape).toStrictEqual({
```

Comment:
Question: What is this test checking? is it that the structure of the landscape data has all the fields defined even if they don't come from the API?

Reply:
yes, and that the code doesn't crash if the fields are missing from the API (this test would've crashed the code before the fix).